exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java

```diff
@@ -148,29 +154,59 @@ private Double getDistinctRowCount(DrillScanRelBase scan, RelMetadataQuery mq, D
       // Statistics cannot be obtained, use default behaviour
       return scan.estimateRowCount(mq) * 0.1;
     }
-    double s = 1.0;
 
-    for (int i = 0; i < groupKey.length(); i++) {
-      final String colName = type.getFieldNames().get(i);
-      // Skip NDV, if not available
-      if (!groupKey.get(i)) {
-        continue;
+    double s = 1.0;
+    if (((int)PrelUtil.getPlannerSettings(
+        scan.getCluster().getPlanner()).getStatisticsJoinCardinalityMode() & 2) == 2) {
+      for (int i = 0; i < groupKey.length(); i++) {
+        final String colName = type.getFieldNames().get(i);
+        // Skip NDV, if not available
+        if (!groupKey.get(i)) {
+          continue;
+        }
+        ColumnStatistics columnStatistics = tableMetadata != null ? tableMetadata.getColumnStatistics(SchemaPath.getSimplePath(colName)) : null;
+        Double ndv = columnStatistics != null ? (Double) columnStatistics.getStatistic(ColumnStatisticsKind.NVD) : null;
+        if (ndv == null) {
+          continue;
+        }
+        s *= ndv;
+        selectivity = getPredSelectivityContainingInputRef(predicate, i, mq, scan);
```

Comment:
Add a comment here regarding the fact that combined NDV that was computed previously is getting scaled down by the predicate selectivity.  Also would be useful to consider the example:  'select a, b from t where a = 10 group by a, b';   Here NDV(a) is affected by the predicate, but NDV(b) is not.

Reply:
Done